forem/forem
# Add ability to mark billboard (display_ad) as "priority" (boolean)

## What type of PR is this? (check all applicable)


## Description

Adds the functionality to mark certain billboards as priority to effectively have them sticky in the "seldom" portion of the final return instead of fade after 1000 views. This has a useful control corollary functionality in the Forem Tools app which will use this API.

### Optimization considerations

I don't think there is much benefit in indexing this boolean field, as it has low cardinality. Since relatively few elements will be marked as priority, there is _maybe_ an argument for indexing, but I think that no index is fine.

I think this `OR` query is appropriate. I think in some scenarios a `UNION` could be more performant, but I think all that is overthinking it for now.

I did notice that there are some pre-existing fields we maybe should have an index on, but that's another issue. 

## Related Tickets & Documents



- Closes https://github.com/forem/forem/issues/19565

## QA Instructions, Screenshots, Recordings

Inspect that the API looks good, and that the admin flow works as expected.

Eyeball the query to make sure it passes muster.

## Added/updated tests?

      have not been included_

## [optional] What gif best describes this PR or how it makes you feel?

![first priorities](https://media.giphy.com/media/DpSoj00vkPpswwZU9o/giphy.gif)
